webblog/frontend/src/components/layout/Header.js

```diff
@@ -0,0 +1,80 @@
+import React, {Component} from "react";
+import {Link} from 'react-router-dom';
+import {logout} from "../../actions/auth";
+import {connect} from "react-redux";
+import PropTypes from "prop-types";
+
+export class Header extends Component {
+    static propTypes = {
+        auth: PropTypes.object.isRequired,
+        logout: PropTypes.func.isRequired
+    };
+
+    render() {
+        const {isAuthenticated, user} = this.props.auth;
+
+        const authLinks = (
+            <ul className="navbar-nav ml-auto mt-2 mt-lg-0">
+        <span className="navbar-text mr-3">
+          <strong>{user ? `Welcome ${user.username}` : ""}</strong>
+        </span>
+                <li className="nav-item">
+                    <button
+                        onClick={this.props.logout}
+                        className="nav-link btn btn-info btn-sm text-light"
+                    >
+                        Logout
+                    </button>
+                </li>
+            </ul>
+        );
+
+        const guestLinks = (
```

Comment:
What's the reason of using guestLinks variable ? did you just copy paste a tutorial to get this task done ?

Reply:
I took instructions from the internet. I just separated this code into two parts to reduce redundancy.